src/components/engine.cppm

```diff
@@ -12,4 +13,9 @@ export namespace ecs::components::engine {
         bool enabled;
         float speed;
     };
+
+    struct hitbox {
+        abstractions::rectangle<float> box;
```

Comment:
Other cosmetic remark, that field should rather be named area, since you often take that component as a parameter named `box` and `box.box` is far from a clear access to the hitbox's area, IMHO.